server/service/src/programs/encounter/insert.rs

```diff
@@ -107,7 +82,38 @@ pub fn insert_encounter(
                     }
                 })?;
 
-            Ok(result)
+            if is_latest_doc(ctx, service_provider, &document.name, document.datetime)
```

Comment:
Hmm, is there a way to `update` existing document ? I mean wouldn't is_latest_doc always be true ? Or is it for sync, or extra safety ?

Reply:
There could be of some race condition where two users are editing a document at the same time, e.g. client 1 edits a document but the document only gets send to the server with a delay. In the meantime client 2 could submit a newer document.

Also in sync, documents could come out of order and only the latest document should update the encounter.